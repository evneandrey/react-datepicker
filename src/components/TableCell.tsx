import * as React from 'react';
import { ifExistCall } from '../utils/FunctionUtil';

interface Props {
  className?: string;
  text?: string;
  subText?: string;
  onClick?: (text: string) => void;
  onMouseOver?: (text: string) => void;
}

const Cell: React.FunctionComponent<Props> = ({ className, text, subText, onClick, onMouseOver }) => {
  return (
    <td
      onClick={() => ifExistCall(onClick, text)}
      onMouseOver={() => ifExistCall(onMouseOver, text)}
      className={className}
    >
      <div>{text}</div>
      {subText && <span className="sub__text">{subText}</span>}
    </td>
  );
};

export default Cell;
