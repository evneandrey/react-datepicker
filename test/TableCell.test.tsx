import * as React from 'react';
import * as sinon from 'sinon';
import { shallow, ShallowWrapper } from 'enzyme';
import TableCell from '../src/components/TableCell';

describe('<TableCell/>', () => {
  let component: ShallowWrapper<React.Component>;
  let onClick: sinon.SinonSpy;
  let onMouseOver: sinon.SinonSpy;
  beforeEach(() => {
    onClick = sinon.spy();
    onMouseOver = sinon.spy();
    component = shallow(
      <TableCell text="test" subText="test" className="test" onClick={onClick} onMouseOver={onMouseOver} />
    );
    jest.resetModules();
  });

  it('renders with no props', () => {
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('props text correctly', () => {
    expect(component.find('div').text()).toEqual('test');
  });

  it('props onClick correctly', () => {
    component.simulate('click');
    expect(onClick).toHaveProperty('callCount', 1);
  });

  it('props onMouseOver correctly', () => {
    component.simulate('mouseover');
    expect(onMouseOver).toHaveProperty('callCount', 1);
  });

  it('props subText correctly', () => {
    expect(component.find('span')).toHaveLength(1);
    expect(component.find('span').text()).toEqual('test');
  });

  it('props className correctly', () => {
    expect(component.hasClass('test')).toBeTruthy();
  });
});
