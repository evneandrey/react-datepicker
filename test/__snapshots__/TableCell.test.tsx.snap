// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableCell/> renders with no props 1`] = `
<td
  className="test"
  onClick={[Function]}
  onMouseOver={[Function]}
>
  <div>
    test
  </div>
  <span
    className="sub__text"
  >
    test
  </span>
</td>
`;
